{
    "cveId": "CVE-2020-24340",
    "version": "1.0.0",
    "timestamp": "2024-12-13T17:34:55.850537+00:00",
    "description": "An issue was discovered in picoTCP and picoTCP-NG through 1.7.0. The code that processes DNS responses in pico_mdns_handle_data_as_answers_generic() in pico_mdns.c does not check whether the number of answers/responses specified in a DNS packet header corresponds to the response data available in the packet, leading to an out-of-bounds read, invalid pointer dereference, and Denial-of-Service.",
    "keyphrases": {
        "rootcause": "",
        "weakness": "out-of-bounds read",
        "impact": "Denial-of-Service",
        "vector": "",
        "attacker": "",
        "product": "picoTCP",
        "version": "through 1.7.0",
        "component": "pico_mdns_handle_data_as_answers_generic() in pico_mdns.c"
    }
}
